rdrf/rdrf/testing/unit/tests.py

```diff
@@ -1742,3 +1742,88 @@ def test_hard_delete_patient_and_clinicaldata_sanity_check(self):
         self.assertEqual(patient_model2.active, True)
         self.assertEqual(clinicaldata_model2.active, True)
         self.assertEqual(patient_model2.id, clinicaldata_model2.django_id)
+
+class UpdateCalculatedFieldsTestCase(FormTestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        def form_value(form_name, section_code, cde_code, mongo_record):
+            for form in mongo_record["forms"]:
+                if form["name"] == form_name:
+                    for section in form["sections"]:
+                        if section["code"] == section_code:
+                            for cde in section["cdes"]:
+                                if cde["code"] == cde_code:
+                                    return cde["value"]
+        self.form_value = form_value
+
+        from rdrf.management.commands.update_calculated_fields import context_ids_for_patient_and_form
+        context_ids = context_ids_for_patient_and_form(self.patient, self.simple_form.name, self.registry)
+        self.context_id = context_ids[0]
+
+        ff = FormFiller(self.simple_form)
+        ff.sectionA.CDEName = "Fred"
+        ff.sectionA.CDEAge = 20
+        ff.sectionB.CDEHeight = 1.82
+        ff.sectionB.CDEWeight = 86.0
+        ff.sectionB.CDEBMI = 38
+
+        form_data = ff.data
+        request = self._create_request(self.simple_form, form_data)
```

Comment:
This a fixed calculation , would it be possible to have base class CalculatedFieldTestCase  ?

Reply:
we have CalculatedFunctionsTestCase that contains all calculated function tests.